### DISCUSSION TITLE
Beatbump is still alive!

### DISCUSSION THREAD

**Discussion by snuffyDev** [snuffyDev/Beatbump#164] [Announcements]
Hey all! Just wanted to give you guys an update on the status of Beatbump. I know there hasn't been much in the way of commits, so it doesn't feel like the project is actively maintained -- but trust me, it is!

## What's been up

I've been really busy dealing with a few big challenges that life threw my way, and also did a lot of work for my job. Still working on the challenges, but my work schedule is lighter (if my boss is reading this, I appreciate it for many reasons! Thanks again!)

There has been a few pull requests over the past few months (thank you again!) and because of those I realized the codebase is still a mess. Although it is better than before, it's still very much a house of cards it feels.

## What's happening 

I am working on doing yet another refactor and cleanup, and tackling a few Issues that were made, and hopefully this will benefit everyone: users, contributors, and myself.

## How things will change

A few months ago I took a step in the right direction with the API and made it more 'constrained'. What I mean by this is within `app/src/routes/api/_lib/request.ts` there's a function called `buildAPIRequest` that is fully typed and offers a 'standardized' way of making the HTTP requests to YTM.

But that was a few months ago, and a single step just isn't good enough. At the time of writing, theres a few API-related issues, the most obvious one being [all of the "Genre" pages](https://beatbump.ml/explore/ggMPOg1uX1NPTld3SDN3WGs4). 

Here's the parsing code for that endpoint:

<figure>
<img src="https://user-images.githubusercontent.com/72365477/233872503-7f69a614-3abc-4af0-bb48-2d9cffcec9f8.png"/>
	<figcaption><p align="center"><em>On the left is the current code, on the right are my un-published changes</p></em></figcaption>
</figure>

The fix for this is simple - just parse all the data together and then transform the data based on what "kind" of object it is. This fix should (and will) be done project-wide. Especially considering how many pages would benefit from this change. Theres a few spots in the API that are pretty similar to the code above. Those spots are where things can, and usually do, break.

Along with this, the components are getting a total revamp (potentially the UI will be changed as well).

Hope y'all will be excited to see this! If anyone has any suggestions, ideas, or questions - open an issue or comment on this!

Thank you for your support in making Beatbump into what it has become!

**Comment by davidcollini** [comment]
One of the greatest joys in life is receiving an update on beatbump's development after a long break

**Reply by snuffyDev** [reply]
@IronMaltese First commit for the refactor was made earlier today in the `dev` branch!

**Comment by snuffyDev** [comment]
AFAICT this works better than prod. already
https://user-images.githubusercontent.com/72365477/235375163-90af4ea3-4c6c-49b1-8000-6782d83948f6.mp4

**Comment by snuffyDev** [comment]
There's a preview deploy live now: https://beatbump-preview.beatbump.workers.dev/home

The backend rewrite is nearly complete! Soon moving onto the frontend

https://user-images.githubusercontent.com/72365477/236613398-0a14735a-ad86-4ea9-8976-672fd8d63af5.mp4
